device-data/types/basal/suppressed.md

```diff
@@ -1,5 +1,279 @@
+  "uploadId": "SampleUploadId"
+}
+```
+
+The three-hour `temp` basal will cross schedule boundaries at 1 a.m. and 3 a.m., and so it will end up being divided into three segment intervals with a `suppressed` to match the segment of the schedule that would have been in effect at that time if the `temp` had not been programmed.
+
+First `temp` interval:
+
+```json
+{
+  "type": "basal",
+  "deliveryType": "temp",
+  "duration": 2100000, // 35 minutes from 12:25 a.m. to 1:00 a.m.
+  "percent": 0.5,
+  "rate": 0.125, // == percent * suppressed.rate
+  "suppressed": {
+    "type": "basal",
+    "deliveryType": "scheduled",
+    "rate": "0.25",
+    "scheduleName": "Standard"
+  },
+  "clockDriftOffset": 0,
+  "conversionOffset": 0,
+  "deviceId": "DevId0987654321",
+  "deviceTime": "2016-10-07T00:25:00",
+  "time": "2016-10-07T07:25:00.000Z",
+  "timezoneOffset": -420,
+  "uploadId": "SampleUploadId"
+}
+```
+
+Second `temp` interval:
+
+```json
+{
+  "type": "basal",
+  "deliveryType": "temp",
+  "duration": 7200000, // 2 hours from 1:00 a.m. to 3:00 a.m.
+  "percent": 0.5,
+  "rate": 0.1, // == percent * suppressed.rate
+  "suppressed": {
+    "type": "basal",
+    "deliveryType": "scheduled",
+    "rate": "0.2",
+    "scheduleName": "Standard"
+  },
+  "clockDriftOffset": 0,
+  "conversionOffset": 0,
+  "deviceId": "DevId0987654321",
+  "deviceTime": "2016-10-07T01:00:00",
+  "time": "2016-10-07T08:00:00.000Z",
+  "timezoneOffset": -420,
+  "uploadId": "SampleUploadId"
+}
+```
+
+Third `temp` interval:
+
+```json
+{
+  "type": "basal",
+  "deliveryType": "temp",
+  "duration": 1500000, // 25 minutes from 3:00 a.m. to 3:25 a.m.
+  "percent": 0.5,
+  "rate": 0.125, // == percent * suppressed.rate
+  "suppressed": {
+    "type": "basal",
+    "deliveryType": "scheduled",
+    "rate": "0.25",
+    "scheduleName": "Standard"
+  },
+  "clockDriftOffset": 0,
+  "conversionOffset": 0,
+  "deviceId": "DevId0987654321",
+  "deviceTime": "2016-10-07T03:00:00",
+  "time": "2016-10-07T10:00:00.000Z",
+  "timezoneOffset": -420,
+  "uploadId": "SampleUploadId"
+}
+```
+
+The `duration`s of all three `temp` intervals here adds up to the programmed `temp` duration: `2100000 + 7200000 + 1500000 = 10800000` (three hours).
+
+For a `suspend` that crosses `scheduled` boundaries, the examples would be very similar, except with no `rate` on the top-level (active) `suspend` basal.
+
+<!-- TODO: DISCUSS WITH GERRIT AND DARIN!!! -->
+**NB:** A *known* issue with this data model is that when a `temp` or `suspend` basal is programmed for a certain `duration` and crosses more than one schedule boundary but then is *canceled* early within one of the "middle" (not edge) segments, we have no good way to represent the original `expectedDuration` of the *entire* programmed `temp` or `suspend`. Rather, the `expectedDuration` on a middle segment of a three-or-more segment `temp` or `suspend` basal should be the expected `duration` of *that* segment from the basal schedule.
```

Comment:
To close this loop, I will add a proposal into the "20160815 Proposed Data Schema Changes" document about potentially adding a new event type that describes the original "user action" to set a temporary basal. It will include the more immediate proposal to add this information into the `payload` of the first segment of the temporary basal.